src/main/java/uk/gov/companieshouse/documentstore/consumer/logging/LoggingKafkaListenerAspect.java

```diff
@@ -21,19 +24,27 @@ class LoggingKafkaListenerAspect {
     private static final String LOG_MESSAGE_RECEIVED = "Processing delta";
     private static final String LOG_MESSAGE_PROCESSED = "Processed delta";
     private static final String EXCEPTION_MESSAGE = "%s exception thrown";
+    private final int maxAttempts;
+
+    LoggingKafkaListenerAspect(@Value("${consumer.max-attempts}") int maxAttempts) {
+        this.maxAttempts = maxAttempts;
+    }
 
     @Around("@annotation(org.springframework.kafka.annotation.KafkaListener)")
     public Object manageStructuredLogging(ProceedingJoinPoint joinPoint) throws Throwable {
 
+        int retryCount = 0;
         try {
             Message<?> message = (Message<?>) joinPoint.getArgs()[0];
+            retryCount = Optional.ofNullable((Integer) joinPoint.getArgs()[1]).orElse(1) - 1;
             DataMapHolder.initialise(extractContextId(message.getPayload())
                     .orElse(UUID.randomUUID().toString()));
 
             DataMapHolder.get()
-                    .topic((String) message.getHeaders().get("kafka_receivedTopic"))
-                    .partition((Integer) message.getHeaders().get("kafka_receivedPartitionId"))
-                    .offset((Long) message.getHeaders().get("kafka_offset"));
+                    .retryCount(retryCount)
```

Comment:
As mentioned above, these should be pulled from the headers rather than add unused args to the wrapped method

Reply:
Done, like Chris' PR for https://github.com/companieshouse/acsp-profile-delta-consumer/pull/6/commits/47fb245bd574e28fdc2529cf97c62c7d8bad0a15